SCons/Script/SConsOptions.py

```diff
@@ -289,14 +289,37 @@ def format_help(self, formatter):
         return result
 
 
+class SConsBadOptionError(optparse.BadOptionError):
```

Comment:
I've always been unclear on whether exceptions should be at point of use or all in `Errors.py` (same for warnings and `Warnings.py`).  Admittedly, this one is odd because it derives from neither a builtin Python exception nor from another SCons-defined exception, but from a stdlib one....